### PR TITLE
Added null safety

### PR DESCRIPTION
**Pull request by Samdid**
## Related Issue 

- Null Safety - I have to migrate the current non-nullable project to the nullable project  

Closes: #37 

### Describe the changes you've made

I have migrated the non-nullable project to nullable project by upgrading the SDK version and some packages. Some packages were non-compatible to null safety, then I have upgraded them by downloading the latest version of that package which is compatible to null safety. Also, changed some code manually to make the project nullable.

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mention any unusual behaviour of your code (Write NA if not)
NA

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly whereever it was hard to understand.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Additional Info (optional)
None

### CODE REVIEW

**Comment by Samdid** [comment]
@shagun25 I have migrated the complete project, please review and merge this PR.

File: Sheheroes_WatchOS/pubspec.lock [hunk 1]
```diff
@@ -0,0 +1,852 @@
+# Generated by pub
+# See https://dart.dev/tools/pub/glossary#lockfile
```
Comment (hiashutoshsingh) [inline Sheheroes_WatchOS/pubspec.lock:2]: Please remove this lock file from commit, but keep this in your local.
See Why: https://dart.dev/guides/libraries/private-files#pubspeclock

File: my_cli/.gitignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Files and directories created by pub
```
Comment (hiashutoshsingh) [inline my_cli/.gitignore:1]: Please remove cli folder, there is no need for this I guess.

File: my_cli/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## 1.0.0
+
```
Comment (hiashutoshsingh) [inline my_cli/CHANGELOG.md:2]: Please remove cli folder.

File: my_cli/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+A sample command-line application with an entrypoint in `bin/`, library code
+in `lib/`, and example unit test in `test/`.
```
Comment (hiashutoshsingh) [inline my_cli/README.md:2]: Please remove cli folder.

File: my_cli/analysis_options.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Defines a default set of lint rules enforced for
+# projects at Google. For details and rationale,
+# see https://github.com/dart-lang/pedantic#enabled-lints.
```
Comment (hiashutoshsingh) [inline my_cli/analysis_options.yaml:3]: Please remove cli folder.

File: my_cli/bin/my_cli.dart [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import 'package:my_cli/my_cli.dart' as my_cli;
+
```
Comment (hiashutoshsingh) [inline my_cli/bin/my_cli.dart:2]: Please remove cli folder.

File: my_cli/lib/my_cli.dart [hunk 7]
```diff
@@ -0,0 +1,3 @@
+int calculate() {
```
Comment (hiashutoshsingh) [inline my_cli/lib/my_cli.dart:1]: Please remove cli folder.

File: my_cli/pubspec.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+name: my_cli
+description: A sample command-line application.
+# version: 1.0.0
```
Comment (hiashutoshsingh) [inline my_cli/pubspec.yaml:3]: Please remove cli folder.

File: my_cli/pubspec.lock [hunk 9]
```diff
@@ -0,0 +1,334 @@
+# Generated by pub
```
Comment (hiashutoshsingh) [inline my_cli/pubspec.lock:1]: Please remove cli folder.

File: my_cli/test/my_cli_test.dart [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import 'package:my_cli/my_cli.dart';
+import 'package:test/test.dart';
+
```
Comment (hiashutoshsingh) [inline my_cli/test/my_cli_test.dart:3]: Please remove cli folder.

File: pubspec.lock [hunk 11]
```diff
@@ -0,0 +1,992 @@
+# Generated by pub
+# See https://dart.dev/tools/pub/glossary#lockfile
```
Comment (hiashutoshsingh) [inline pubspec.lock:2]: Please remove this lock file from commit, but keep this in your local.
See Why: https://dart.dev/guides/libraries/private-files#pubspeclock

**Review by hiashutoshsingh — CHANGES_REQUESTED** [review]
Hey, @Samdid thanks for submitting PR.
This PR needs a lot of changes.
1) I can see a change on 2 files only, the rest files should adhere to null safety.
2) can't see the pubspec.yaml changes for packages.
3) unnecessary cli folder is added by mistake


Please resolve these changes, I am sure you will able to pull this off. 
Let me know if you face any blocker.

**Comment by Samdid** [comment]
@hiashutoshsingh I will resolve all the changes asap

**Comment by hiashutoshsingh** [comment]
Hey @Samdid what's the update on the above ones, this is pending for a long time. Let's try to resolve all commented issues in 2-3 days, I am sure you will be able to pull this off.

**Comment by Samdid** [comment]
> Hey @Samdid what's the update on the above ones, this is pending for a long time. Let's try to resolve all commented issues in 2-3 days, I am sure you will be able to pull this off.

Okay

**Comment by hiashutoshsingh** [comment]
> > Hey @Samdid what's the update on the above ones, this is pending for a long time. Let's try to resolve all commented issues in 2-3 days, I am sure you will be able to pull this off.
> 
> Okay

Hey @Samdid what is the status of the above?

**Comment by hiashutoshsingh** [comment]
> Hey @Samdid what's the update on the above ones, this is pending for a long time. Let's try to resolve all commented issues in 2-3 days, I am sure you will be able to pull this off.

any updates on the above will be closing this PR, wanted to give chance to others.

**Pull request closed without merging**